src/Grpc/JsonTranscoding/src/Microsoft.AspNetCore.Grpc.JsonTranscoding/Internal/JsonTranscodingRouteAdapter.cs

```diff
@@ -169,15 +179,43 @@ public static JsonTranscodingRouteAdapter Parse(HttpRoutePattern pattern)
                         break;
                     case SegmentType.CatchAll:
                         // Ignore remaining segment values.
-                        tempSegments[i] = $"{{**__Discard_{i}}}";
+                        if (pattern.Verb != null)
+                        {
+                            tempSegments[i] = $"{{**__Discard_{i}:regex({Regex.Escape($":{pattern.Verb}")}$)}}";
```

Comment:
Assuming this is not a new usage of `Regex`.

Reply:
What do you mean?

The route adapter added regex constraints before when required. This particular one is new because content after a last segment discard wasn't possible before. Now, verb can come after it and needs to be required with regex constraint.

Reply:
I was thinking in the context of AOT. No issue with it if we are already using Regex in the gRPC side of things.

Reply:
This is gRPC transcoding - https://learn.microsoft.com/en-us/aspnet/core/grpc/json-transcoding?view=aspnetcore-7.0. It's quite different from regular gRPC and it isn't currently compatible with AOT. Even if it is, using regex is fine when required.

Reply:
Does this mean that AddJsonTranscoding should call AddRouting like AddMvcCore does? That way the regexes don't break when combined with CreateSlimBuilder.